kit/Kit.cpp

```diff
@@ -2422,17 +2425,16 @@ int pollCallback(void* pData, int timeoutUs)
         if (p)
             v.push_back(p);
     }
-    lock.unlock();
     if (v.empty())
     {
-        // On iOS, we cannot wait very long otherwise we are waiting here
-        // when the app opens a document on another thread.
-        // TODO: replace sleeping with a wait condition that can be
-        // interrupted if the user initiates opening a document.
-        std::this_thread::sleep_for(std::chrono::seconds(5));
+        // Remove any stale elements from KitSocketPoll::KSPolls and
+        // block until an element is added to KitSocketPoll::KSPolls
+        KitSocketPoll::KSPolls.clear();
```

Comment:
Seems no elements are ever removed from KitSocketPoll::KSPolls. So my assumption here is that if a call to lock() on an element returns a nullptr, it must be a stale entry.